src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -165,7 +165,14 @@ static void OnRegister(DNSServiceRef sdRef, DNSServiceFlags flags, DNSServiceErr
                     Error::ToString(err));
 
     auto sdCtx = reinterpret_cast<RegisterContext *>(context);
-    sdCtx->Finalize(err);
+    VerifyOrReturn(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
+
+    // Once a service has been properly published it is normally unreachable because the hostname has not yet been
+    // registered against the dns daemon. Register the records mapping the hostname to our IP.
+    auto error = sdCtx->mHostNameRegistrar.Register(^(DNSServiceErrorType registerRecordError) {
+        sdCtx->Finalize(registerRecordError);
```

Comment:
So if sdCtx dies before anything is registered... when the timeout fires, won't this be use-after-free?

Look like the bot is catching this too.

Reply:
Thanks !